cpp/src/rolling/rolling_detail.hpp

```diff
@@ -41,7 +41,7 @@ static constexpr bool is_rolling_supported()
       (op == aggregation::SUM) or (op == aggregation::MIN) or (op == aggregation::MAX) or
       (op == aggregation::COUNT_VALID) or (op == aggregation::COUNT_ALL) or
       (op == aggregation::MEAN) or (op == aggregation::ROW_NUMBER) or (op == aggregation::LEAD) or
-      (op == aggregation::LAG);
+      (op == aggregation::LAG) or (op == aggregation::COLLECT);
```

Comment:
This `is_rolling_supported` function is not sustainable. It reveals the fact that more and more functionality has simply been tacked onto the "rolling" algorithm instead of being holistically designed to support all the different kinds of operations. 

When working on the next rolling functionality, concerted effort should be spent on refactoring the design of the rolling interfaces and implementation.

Reply:
Agreed. I wonder if `window_aggregation` should be an abstraction separate from regular `aggregation`. I believe that's how it was when `rolling_window()` first came to be.

Reply:
See https://github.com/rapidsai/cudf/issues/7106

Reply:
Resolving this now. I'll take a crack at #7106 when/before I work on the `RANK()` window function.